### PR TITLE
Update etcdutl restore to create v2 snapshot from v3

### PR DESCRIPTION
**Pull request by geetasg**
Related to https://github.com/etcd-io/etcd/issues/12913
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: tests/e2e/ctl_v3_snapshot_test.go [hunk 1]
```diff
@@ -380,12 +380,16 @@ func TestRestoreCompactionRevBump(t *testing.T) {
...
 	epc.Procs[0].Config().DataDirPath = newDataDir
+
 	for i := range epc.Procs[0].Config().Args {
 		if epc.Procs[0].Config().Args[i] == "--data-dir" {
 			epc.Procs[0].Config().Args[i+1] = newDataDir
 		}
 	}
 
+	//Verify that initial snapshot is created by the restore operation
+	verifySnapshot(t, epc)
```
Comment (ahrtr) [inline tests/e2e/ctl_v3_snapshot_test.go:391]: Can you also add a verification that the membership data read from the v2 snapshot is expected?

Comment (geetasg) [reply tests/e2e/ctl_v3_snapshot_test.go]: updated.

File: tests/e2e/cluster_downgrade_test.go [hunk 2]
```diff
@@ -293,6 +297,23 @@ func verifySnapshot(t *testing.T, epc *e2e.EtcdProcessCluster) {
...
+		st := v2store.New(etcdserver.StoreClusterPrefix, etcdserver.StoreKeysPrefix)
+		err = st.Recovery(snap.Data)
+		assert.NoError(t, err)
+		for _, m := range expectedMembers.Members {
+			_, err := st.Get(membership.MemberStoreKey(types.ID(m.ID)), true, true)
+			assert.NoError(t, err)
+		}
+		t.Logf("Verifed snapshot for member %d", i)
+	}
+	t.Logf("All members have a valid snapshot")
```
Comment (ahrtr) [inline tests/e2e/cluster_downgrade_test.go:314]:
```suggestion
	t.Log("All members have a valid snapshot")
```

**Review by ahrtr — APPROVED** [review]
LGTM with a nit.

thx

File: tests/e2e/ctl_v3_snapshot_test.go [hunk 3]
```diff
@@ -380,12 +383,16 @@ func TestRestoreCompactionRevBump(t *testing.T) {
...
 	t.Log("(Re)starting the etcd member using the restored snapshot...")
 	epc.Procs[0].Config().DataDirPath = newDataDir
+
 	for i := range epc.Procs[0].Config().Args {
 		if epc.Procs[0].Config().Args[i] == "--data-dir" {
 			epc.Procs[0].Config().Args[i+1] = newDataDir
 		}
 	}
 
+	//Verify that initial snapshot is created by the restore operation
```
Comment (jmhbnz) [inline tests/e2e/ctl_v3_snapshot_test.go:393]: Nit: Space after starting comment to align with style used for etcd.

```suggestion
	// Verify that initial snapshot is created by the restore operation
```

**Review by jmhbnz — APPROVED** [review]
Overall lgtm, nice work @geetasg 👍🏻

File: tests/e2e/ctl_v3_snapshot_test.go [hunk 4]
```diff
@@ -380,12 +383,16 @@ func TestRestoreCompactionRevBump(t *testing.T) {
...
 	epc.Procs[0].Config().DataDirPath = newDataDir
+
 	for i := range epc.Procs[0].Config().Args {
 		if epc.Procs[0].Config().Args[i] == "--data-dir" {
 			epc.Procs[0].Config().Args[i+1] = newDataDir
 		}
 	}
 
+	// Verify that initial snapshot is created by the restore operation
+	verifySnapshotMembers(t, epc, membersBefore)
```
Comment (ahrtr) [inline tests/e2e/ctl_v3_snapshot_test.go:394]: The cluster in this test case has only one member,  please consider to add a case verify that the members read from the snapshot are also expected in multiple members cluster in a separate PR.

**Pull request merged**